AWS CodeBuild artifact location should allow path per documentation

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
Terraform v0.11.8
+ provider.aws v1.40.0
```

### Terraform Configuration Files
This is the relevant part of the config, it is used inside a module:
```hcl
variable "deployment_bucket" {
  default = "some-bucket"
}

resource "aws_codebuild_project" "test" {
  artifacts {
    type = "S3"
    location = "${var.deployment_bucket}/some-path/whatever"
  }
  ...
}
```

### Debug Output
This is the relevant part of the output:
```
2018/10/11 18:08:24 [ERROR] root.codebuild_prod: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* aws_codebuild_project.test: Error creating CodeBuild project: InvalidInputException: Invalid artifacts: location should not have a forward-slash
        status code: 400, request id: e1baf54e-cd6f-11e8-bc04-03e2c94a323d
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: 2018/10/11 18:08:24 [DEBUG] [aws-sdk-go] DEBUG: Response codebuild/CreateProject Details:
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: ---[ RESPONSE ]--------------------------------------
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: HTTP/1.1 400 Bad Request
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: Connection: close
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: Content-Length: 106
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: Content-Type: application/x-amz-json-1.1
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: Date: Thu, 11 Oct 2018 16:08:24 GMT
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: X-Amzn-Requestid: e1bc5486-cd6f-11e8-abae-db8fd2383e9b
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: 
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: 
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: -----------------------------------------------------
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: 2018/10/11 18:08:24 [DEBUG] [aws-sdk-go] {"__type":"InvalidInputException","message":"Invalid artifacts: location should not have a forward-slash"}
2018-10-11T18:08:24.802+0200 [DEBUG] plugin.terraform-provider-aws_v1.40.0_x4: 2018/10/11 18:08:24 [DEBUG] [aws-sdk-go] DEBUG: Validate Response codebuild/CreateProject failed, not retrying, error InvalidInputException: Invalid artifacts: location should not have a forward-slash

```

### Expected Behavior
The [documentation](https://www.terraform.io/docs/providers/aws/r/codebuild_project.html#location) states:

> (Optional) Information about the build output artifact location. If type is set to CODEPIPELINE or NO_ARTIFACTS then this value will be ignored. **If type is set to S3, this is the name of the output bucket. If path is not also specified, then location can also specify the path of the output artifact in the output bucket.**

Hence, the `location` (interpolated) value `some-bucket/some-path/whatever` should be valid.

### Actual Behavior
The apply fails with:

> Invalid artifacts: location should not have a forward-slash

### Steps to Reproduce
1. `terraform init`
2. `terraform apply`

--- a/website/docs/r/codebuild_project.html.markdown
+++ b/website/docs/r/codebuild_project.html.markdown
@@ -235,7 +235,7 @@ The following arguments are supported:
 * `type` - (Required) The build output artifact's type. Valid values for this parameter are: `CODEPIPELINE`, `NO_ARTIFACTS` or `S3`.
 * `encryption_disabled` - (Optional) If set to true, output artifacts will not be encrypted. If `type` is set to `NO_ARTIFACTS` then this value will be ignored. Defaults to `false`.
 * `override_artifact_name` (Optional) If set to true, a name specified in the build spec file overrides the artifact name.
-* `location` - (Optional) Information about the build output artifact location. If `type` is set to `CODEPIPELINE` or `NO_ARTIFACTS` then this value will be ignored. If `type` is set to `S3`, this is the name of the output bucket. If `path` is not also specified, then `location` can also specify the path of the output artifact in the output bucket.
+* `location` - (Optional) Information about the build output artifact location. If `type` is set to `CODEPIPELINE` or `NO_ARTIFACTS` then this value will be ignored. If `type` is set to `S3`, this is the name of the output bucket.
 * `name` - (Optional) The name of the project. If `type` is set to `S3`, this is the name of the output artifact object
 * `namespace_type` - (Optional) The namespace to use in storing build artifacts. If `type` is set to `S3`, then valid values for this parameter are: `BUILD_ID` or `NONE`.
 * `packaging` - (Optional) The type of build output artifact to create. If `type` is set to `S3`, valid values for this parameter are: `NONE` or `ZIP`

